### PR TITLE
Fix type-changing moves & abilities change while Dynamaxed

### PR DESCRIPTION
**Pull request by 420Blazeitt**
Probably not the cleanest way to do it but it works ^^'

### CODE REVIEW

**Comment by Zarel** [comment]
I'm not particularly fond of how hardcoded this is.

**Comment by Zarel** [comment]
Do these all run the same type-changing event?

**Comment by 420Blazeitt** [comment]
yeah i’m sorry, uh. They all have onModifyMove. But that’s about it

File: data/scripts.js [hunk 1]
```diff
@@ -161,6 +161,15 @@ let BattleScripts = {
...
 			move = this.getActiveZMove(move, pokemon);
 		}
+		if (pokemon.hasAbility('pixilate') || pokemon.hasAbility('refrigerate') || pokemon.hasAbility('galvanize') || pokemon.hasAbility('normalize') || pokemon.hasAbility('liquidvoice') && maxMove) {
+			this.singleEvent('ModifyMove', pokemon.getAbility(), null, pokemon, target, pokemon.getAbility(), move);
+			sourceEffect = move;
+		}
+		if (['aurawheel', 'hiddenpower', 'judgment', 'multiattack', 'naturalgift', 'revelationdance', 'technoblast', 'weatherball'].includes(move.id) && maxMove) {
+			// Max moves change type when used directly
+			this.singleEvent('ModifyMove', move, null, pokemon, target, move, move);
+			if (move.type !== 'Normal') sourceEffect = move;
```
Comment (KrisXV) [inline data/scripts.js]: I didn't know Aura Wheel was a Normal-type type-changing move

**Comment by 420Blazeitt** [comment]
Alright there is no hardcoding used in this way, but still works just as well!

**Comment by Zarel** [comment]
I'd prefer a separate `ModifyType` event, so we don't have to go through the full `ModifyMove` rigamarole.

**Comment by 420Blazeitt** [comment]
yeah that’s a good idea

**Comment by 420Blazeitt** [comment]
I haven't typed anything properly, will get to that later

File: data/scripts.js [hunk 2]
```diff
@@ -155,12 +155,19 @@ let BattleScripts = {
...
-			this.singleEvent('ModifyMove', move, null, pokemon, target, move, move);
+			this.singleEvent('ModifyType', move, null, pokemon, target, move, move);
 			if (move.type !== 'Normal') sourceEffect = move;
 		}
 		if (zMove || (move.category !== 'Status' && sourceEffect && /** @type {ActiveMove} */(sourceEffect).isZ)) {
 			move = this.getActiveZMove(move, pokemon);
 		}
+		if (maxMove && move.category !== 'Status') {
+			let moveType = move.type;
+			// Max move outcome is dependant on the move type after type modifications from ability and the move itself
```
Comment (Zarel) [inline data/scripts.js]:
```suggestion
			// Max move outcome is dependent on the move type after type modifications from ability and the move itself
```

**Comment by Zarel** [comment]
This isn't the signature I'd use, but it's a good first step. I'd merge it after the spelling fix.

**Comment by 420Blazeitt** [comment]
So, I made the typo fix. ;w;

**Comment by Amaterasu5** [comment]
Hey! sorry to revive an old thread but this doesn't appear to be fixed in the damage calc. Using max weather ball in the sun from torkoal, for example, registers in the calc as max strike without the damage or sun boost. I've attached an image:
Screen Shot 2020-08-07 at 9 53 02 PM

**Comment by Zarel** [comment]
@Amaterasu5 you should report that in the damagecalc:

https://github.com/smogon/damage-calc

**Comment by Amaterasu5** [comment]
@Zarel will do, thanks!

**Pull request merged**